src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -394,6 +396,11 @@ public function load(array $configs, ContainerBuilder $container)
             // remove tagged iterator argument for resource checkers
             $container->getDefinition('config_cache_factory')->setArguments([]);
         }
+
+        if ($config['disallow_search_engine_index'] ?? false) {
+            $container->register(DisallowRobotsIndexingListener::class)
+                ->addTag('kernel.event_listener', ['event' => KernelEvents::RESPONSE, 'priority' => -255]);
```

Comment:
The listener configuration should on the listener class instead (via EventSubscriberInterface).

Reply:
Fixed